[UI] Settings > Labels - Filter throws error: description is undefined.  Take a look at "searchKeys".

<!--

Thank you for reporting a bug in InfluxDB. 

* Please ask usage questions on the Influx Community site.
    * https://community.influxdata.com/
* Please add a :+1: or comment on a similar existing bug report instead of opening a new one.
    * https://github.com/influxdata/influxdb/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+is%3Aclosed+sort%3Aupdated-desc+label%3Akind%2Fbug+
* Please check whether the bug can be reproduced with the latest release.
* The fastest way to fix a bug is to open a Pull Request.
    * https://github.com/influxdata/influxdb/pulls

-->

Reviewing Settings > Labels to prepare writing cypress tests. 

__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Open Settings > Labels
2. Add three simple labels e.g. {ASDF, QWERTY, ZCVB} (see screenshots attached below)
3. Click on the Filter field
4. Attempt to type 'ASDF' 

__Expected behavior:__

Expected that the label ASDF would remain visible and the other labels would be hidden.

__Actual behavior:__

Received the error message "An InfluxDB error has occurred. Please report the issue "

N.B. First attempts to filter seemed to work.  At some point the UI entered a state where further attempts ended in error. 

__Environment info:__

* System info: Run `uname -srm` and copy the output here `Linux x86_64`
* InfluxDB version: Run `influxd version` and copy the output here - this is a local build for preparing automated tests.  Latest commit before local merge `e725e1a2bc7361b3cbf3e3b85984cd6cbf1d9db2`
* Other relevant environment details: Container runtime, disk info, etc

Browser: Chrome 73

__Config:__
Copy any non-default config values here or attach the full config as a gist or file.

Boot command 

```
./bin/linux/influxd --store=memory --e2e-testing=true --log-level debug 2>&1 > ~/var/log/influxd.log &
```
<!-- The following sections are only required if relevant. -->
__Screenshots:__
![LabelsFilter01](https://user-images.githubusercontent.com/25881301/57783591-b1ff2480-772e-11e9-994e-3c489f0383ad.png)
![LabelsFilter02](https://user-images.githubusercontent.com/25881301/57783593-b1ff2480-772e-11e9-8b01-73ba19a7dbce.png)


__Logs:__
Include snippet of errors in log.

console log 
```
src.9817a40b.js:694 Error: description is undefined.  Take a look at "searchKeys". 
    at src.9817a40b.js:1823
    at Array.some (<anonymous>)
    at src.9817a40b.js:1823
    at Array.filter (<anonymous>)
    at n.get (src.9817a40b.js:1823)
    at n.get (src.9817a40b.js:1823)
    at n.value (src.9817a40b.js:1823)
    at Ba (src.9817a40b.js:694)
    at ja (src.9817a40b.js:694)
    at $a (src.9817a40b.js:694)
Si @ src.9817a40b.js:694
Oi.a.componentDidCatch.n.callback @ src.9817a40b.js:694
gi @ src.9817a40b.js:694
vi @ src.9817a40b.js:694
no @ src.9817a40b.js:694
ao @ src.9817a40b.js:694
(anonymous) @ src.9817a40b.js:694
exports.unstable_runWithPriority @ src.9817a40b.js:690
Xo @ src.9817a40b.js:694
Yo @ src.9817a40b.js:694
$o @ src.9817a40b.js:694
Oe @ src.9817a40b.js:694
Le @ src.9817a40b.js:694
Rn @ src.9817a40b.js:694
(anonymous) @ src.9817a40b.js:694
exports.unstable_runWithPriority @ src.9817a40b.js:690
eu @ src.9817a40b.js:694
Mn @ src.9817a40b.js:694
src.9817a40b.js:1441 Error: description is undefined.  Take a look at "searchKeys". 
    at src.9817a40b.js:1823
    at Array.some (<anonymous>)
    at src.9817a40b.js:1823
    at Array.filter (<anonymous>)
    at n.get (src.9817a40b.js:1823)
    at n.get (src.9817a40b.js:1823)
    at n.value (src.9817a40b.js:1823)
    at Ba (src.9817a40b.js:694)
    at ja (src.9817a40b.js:694)
    at $a (src.9817a40b.js:694)
value @ src.9817a40b.js:1441
Oi.a.componentDidCatch.n.callback @ src.9817a40b.js:694
gi @ src.9817a40b.js:694
vi @ src.9817a40b.js:694
no @ src.9817a40b.js:694
ao @ src.9817a40b.js:694
(anonymous) @ src.9817a40b.js:694
exports.unstable_runWithPriority @ src.9817a40b.js:690
Xo @ src.9817a40b.js:694
Yo @ src.9817a40b.js:694
$o @ src.9817a40b.js:694
Oe @ src.9817a40b.js:694
Le @ src.9817a40b.js:694
Rn @ src.9817a40b.js:694
(anonymous) @ src.9817a40b.js:694
exports.unstable_runWithPriority @ src.9817a40b.js:690
eu @ src.9817a40b.js:694
Mn @ src.9817a40b.js:694
src.9817a40b.js:1441 {componentStack: "↵    in n↵    in s↵    in Connect(s)↵    in t↵    … in RouterContext↵    in Router↵    in r↵    in o"}
value @ src.9817a40b.js:1441
Oi.a.componentDidCatch.n.callback @ src.9817a40b.js:694
gi @ src.9817a40b.js:694
vi @ src.9817a40b.js:694
no @ src.9817a40b.js:694
ao @ src.9817a40b.js:694
(anonymous) @ src.9817a40b.js:694
exports.unstable_runWithPriority @ src.9817a40b.js:690
Xo @ src.9817a40b.js:694
Yo @ src.9817a40b.js:694
$o @ src.9817a40b.js:694
Oe @ src.9817a40b.js:694
Le @ src.9817a40b.js:694
Rn @ src.9817a40b.js:694
(anonymous) @ src.9817a40b.js:694
exports.unstable_runWithPriority @ src.9817a40b.js:690
eu @ src.9817a40b.js:694
Mn @ src.9817a40b.js:694

```

__Performance:__
Generate profiles with the following commands for bugs related to performance, locking, out of memory (OOM), etc.

Not a DB performance issue.

```sh
# Commands should be run when the bug is actively.
# Note: This command will run for at least 30 seconds.
curl -o profiles.tar.gz "http://localhost:8086/debug/pprof/all?cpu=true"
curl -o vars.txt "http://localhost:8086/debug/vars"
iostat -xd 1 30 > iostat.txt
# Attach the `profiles.tar.gz`, `vars.txt`, and `iostat.txt` output files.
```
However performance files attached as requested. 

[iostat.txt](https://github.com/influxdata/influxdb/files/3183037/iostat.txt)
[profiles.tar.gz](https://github.com/influxdata/influxdb/files/3183038/profiles.tar.gz)
[vars.txt](https://github.com/influxdata/influxdb/files/3183039/vars.txt)

--- a/ui/cypress/e2e/labels.test.ts
+++ b/ui/cypress/e2e/labels.test.ts
@@ -188,7 +188,7 @@ describe('labels', () => {
       "(\u03944) J'entends par attribut ce que l'entendement perçoit d'une substance comme constituant son essence. "
     const newLabelColor = '#B0D0FF'
 
-    //create label
+    // create label
 
     cy.get<Organization>('@org').then(({id}) => {
       cy.createLabel(oldLabelName, id, {
@@ -197,14 +197,16 @@ describe('labels', () => {
       })
     })
 
-    //verify name, descr, color
+    // verify name, descr, color
     cy.getByTestID('label-card').should('have.length', 1)
     cy.getByTestID('label-card')
       .contains(oldLabelName)
       .should('be.visible')
+
     cy.getByTestID('label-card')
       .contains(oldLabelDescription)
       .should('be.visible')
+
     cy.getByTestID('label-card')
       .children('div.resource-list--name-meta')
       .children('div.label')
@@ -219,12 +221,13 @@ describe('labels', () => {
       .children('div')
       .invoke('text')
       .should('equal', 'Edit Label')
-    //dissmiss
+
+    // dismiss
     cy.getByTestID('overlay--header')
       .children('button')
       .click()
 
-    //modify
+    // modify
     cy.getByTestID('label-card')
       .contains(oldLabelName)
       .click()
@@ -240,7 +243,7 @@ describe('labels', () => {
       .type(newLabelColor)
     cy.getByTestID('create-label-form--submit').click()
 
-    //verify name, descr, color
+    // verify name, descr, color
     cy.getByTestID('label-card').should('have.length', 1)
     cy.getByTestID('label-card')
       .contains(newLabelName)

--- a/ui/src/labels/components/LabelsTab.tsx
+++ b/ui/src/labels/components/LabelsTab.tsx
@@ -101,7 +101,7 @@ class Labels extends PureComponent<Props, State> {
         </TabbedPageHeader>
         <FilterList<ILabel>
           list={labels}
-          searchKeys={['name', 'description']}
+          searchKeys={['name', 'properties.description']}
           searchTerm={searchTerm}
         >
           {ls => (

--- a/ui/src/shared/components/Filter.test.tsx
+++ b/ui/src/shared/components/Filter.test.tsx
@@ -173,6 +173,47 @@ describe('FilterList', () => {
     expect(expected[2].textContent).toEqual(itemThree.name)
   })
 
+  it('can filter nested objects', () => {
+    const itemOne = {
+      id: '1',
+      name: 'crackle',
+      properties: {
+        description: 'a',
+      },
+    }
+
+    const itemTwo = {
+      id: '2',
+      name: 'ports',
+      properties: {
+        description: 'b',
+      },
+    }
+
+    const itemThree = {
+      id: '3',
+      name: 'rando',
+      properties: {
+        description: 'z',
+      },
+    }
+
+    const list = [itemOne, itemTwo, itemThree]
+
+    const searchTerm = 'Z'
+    const searchKeys = ['name', 'properties.description']
+    const {getAllByTestId} = setup({
+      list,
+      searchTerm,
+      searchKeys,
+    })
+
+    const expected = getAllByTestId('list-item')
+
+    expect(expected.length).toEqual(1)
+    expect(expected[0].textContent).toEqual(itemThree.name)
+  })
+
   it('errors when searchKey value is an object', () => {
     const itemOne = {
       name: {

--- a/ui/src/shared/components/Filter.tsx
+++ b/ui/src/shared/components/Filter.tsx
@@ -56,7 +56,7 @@ export default class FilterList<T> extends PureComponent<Props<T>> {
 
         if (!isStringArray && _.isObject(value)) {
           throw new Error(
-            `The value at key "${key}" is an object.  Take a look at "searchKeys" and 
+            `The value at key "${key}" is an object.  Take a look at "searchKeys" and
              make sure you're indexing onto a primitive value`
           )
         }
@@ -64,10 +64,6 @@ export default class FilterList<T> extends PureComponent<Props<T>> {
         if (isStringArray) {
           const searchIndex = this.createIndex(value)
           return this.checkIndex(searchIndex, formattedSearchTerm)
-        }
-
-        if (value === '') {
-          throw new Error(`${key} is undefined.  Take a look at "searchKeys". `)
         }
 
         return String(value)

